### ISSUE TITLE
xDAI support

### ISSUE THREAD

**Issue by anxolin** [Uniswap/uniswap-sdk#52]
Hi, give the big traction xDAI network, it's alignment with Ethereum roadmap, and the fact that is fully compatible with Ethereum, could it be supported by the SDK?

These are their endpoints:
https://www.xdaichain.com/for-developers/developer-resources#json-rpc-endpoints

Chain Ids:
https://github.com/Uniswap/uniswap-sdk/blob/v2/src/constants.ts#L6

**Comment by anxolin** [comment]
Also, I just want to add, that right now there's a fork of Uniswap in xDAI called HoneySwap, https://www.xdaichain.com/about-xdai/project-spotlights/1hive#honey-liquidity-and-exchange 

it would be nice to see Uniswap "the original" also in that network, and since it's 100% compatible, I don't think it would be too much development effort. xDAI will allow to swap for less than a cent! Great for many use cases, like being able to use dapps that are prohibitively expensive in mainnet, and onboarding users.

**Comment by anxolin** [comment]
@moodysalem don't you think this would be good for Uniswap?

**Comment by moodysalem** [comment]
It seems like all that is needed is to add a value to the enum, but since it can be any network ID it might be better to support a special OTHER chainId that can be used with any network. I would happily review a PR that adds that.

**Comment by anxolin** [comment]
I went for it, and created the PR #53  @moodysalem

**Comment by moodysalem** [comment]
I've refactored the code to use `ChainId | number` everywhere which only fails if you are trying to swap to/from the `Currency.ETHER` constant (because of how we map between WETH9/Ether)

Please let me know if the SDK is flexible enough for your use case now

**Comment by anxolin** [comment]
@moodysalem is this version available in https://www.npmjs.com/package/@uniswap/sdk?

**Comment by moodysalem** [comment]
> @moodysalem is this version available in https://www.npmjs.com/package/@uniswap/sdk?

No, the package is now `@uniswap/v2-sdk` for consistency with core and periphery

But you might also be blocked by https://github.com/Uniswap/uniswap-v2-sdk/issues/55
